cli/azd/pkg/project/dotnet_importer.go

```diff
@@ -272,9 +275,177 @@ func (ai *DotNetImporter) Services(
 
 		services[svc.Name] = svc
 	}
+
+	buildContainers, err := apphost.BuildContainers(manifest)
+	if err != nil {
+		return nil, err
+	}
+	for name, bContainer := range buildContainers {
+
+		defaultLanguage := ServiceLanguageDotNet
+		relativePath := svcConfig.RelativePath
+		var dOptions DockerProjectOptions
+
+		if bContainer.Build != nil {
+			defaultLanguage = ServiceLanguageDocker
+			relPath, err := filepath.Rel(p.Path, filepath.Dir(bContainer.Build.Dockerfile))
+			if err != nil {
+				return nil, err
+			}
+			relativePath = relPath
+
+			bArgs, err := evaluateArgsWithConfig(*manifest, bContainer.Build.Args)
+			if err != nil {
+				return nil, fmt.Errorf("evaluating build args for service %s: %w", name, err)
+			}
+			bArgsArray, reqEnv, err := buildArgsArrayAndEnv(*manifest, bContainer.Build.Secrets)
+			if err != nil {
+				return nil, fmt.Errorf("converting build args to array for service %s: %w", name, err)
+			}
+
+			dOptions = DockerProjectOptions{
+				Path:         bContainer.Build.Dockerfile,
+				Context:      bContainer.Build.Context,
+				BuildArgs:    mapToStringSlice(bArgs, "="),
+				BuildSecrets: bArgsArray,
+				BuildEnv:     reqEnv,
+			}
+		}
+
+		svc := &ServiceConfig{
+			RelativePath: relativePath,
+			Language:     defaultLanguage,
+			Host:         DotNetContainerAppTarget,
+			Docker:       dOptions,
+		}
+
+		svc.Name = name
+		svc.Project = p
+		svc.EventDispatcher = ext.NewEventDispatcher[ServiceLifecycleEventArgs]()
+
+		svc.Infra.Provider, err = provisioning.ParseProvider(svc.Infra.Provider)
+		if err != nil {
+			return nil, fmt.Errorf("parsing service %s: %w", svc.Name, err)
+		}
+
+		svc.DotNetContainerApp = &DotNetContainerAppOptions{
+			ContainerImage: bContainer.Image,
+			Manifest:       manifest,
+			ProjectName:    name,
+			AppHostPath:    svcConfig.Path(),
+		}
+		services[svc.Name] = svc
+
+	}
 	return services, nil
 }
 
+// buildArgsArray produces an array of args to pass to the container build command.
+// See: https://docs.docker.com/build/building/secrets/
+func buildArgsArrayAndEnv(
```

Comment:
Might be nice to introduce `dotnet_importer_test.go` and write some unit tests for some of this logic.  There's a fair amount of transformations going on here and if we can have some quick tests to alert us when something is wrong, that could be good?  I'm just a little worried about the lack of coverage here (even though I know we'll get some coverage via end to end tests from Aspire in the future).

Reply:
tests added